### PR TITLE
Modify doc index page

### LINKED ISSUE #2783: Docs landing page should showcase examples
This is just a suggestion based on user feedback.

The landing page of the docs (https://nest-simulator.readthedocs.org/) could showcase our coolest examples ("applications") from https://nest-simulator.readthedocs.io/en/v3.4/examples/index.html#pynest-examples, rather than a little script snippet which is typically glossed over (too much technical detail for a landing page). Applications are much more telling about what you can do with NEST (what it can do for you!)

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR is taking over for #2969 but being split into multiple PRs so better organized and easier to review.

This PR improves the overall landing page and structure of contents. 

Resolves #2783 

## The new index page 

- It removes the index.html template in favour of a restructured text page, which will eliminate issues with compatibility between the sphinx_material theme and custom template. 
- contains a 'carousel' where we can highlight new and interesting examples. (TODO, decide on which examples to showcase)
- contains a modified table of contents (TOC)
- contains a conceptual diagram using elements from NEST Desktop and images courtesy of A. Fischer  TODO: Add description to diagram
- A 'get started here' button leads to the new tutorials and guides section, which is what is seen on the TOC 

NOTE changes to other pages will be done in subsequent PRs! The modifications done for tutorial and guide index are interim changes.

### CODE REVIEW

File: doc/htmldoc/conf.py [hunk 1]
```diff
@@ -177,7 +181,18 @@
...
+    "css/custom.css",
+    "css/popup.css",
+    "css/pygments.css",
+]
+
+html_js_files = [
+    "js/custom.js",
+    "js/popup.js",
+]
+# html_additional_pages = {"index": "index.html"}
```
Comment (clinssen) [inline doc/htmldoc/conf.py:195]: remove?

File: doc/htmldoc/static/css/custom.css [hunk 2]
```diff
@@ -8,21 +8,23 @@
...
    Sphinx design properties use the sd in variable names
  */
 /* Define colors*/
 :root {
     --nest-orange: #ff6633;
-    --nest-blue: #1281B3;
-    --nest-green: #36B34F;
+    --nest-blue: #0E6A93;
+    --nest-dkblue: #072f42;
+    --nest-brown: #4b210c; /*formally green */
```
Comment (clinssen) [inline doc/htmldoc/static/css/custom.css:21]: ?

File: doc/htmldoc/static/css/custom.css [hunk 3]
```diff
@@ -115,7 +113,11 @@ rect.basic.label-container {
 	stroke: #ff6633;
 	stroke-width: 1px;
 }
-
+/*
```
Comment (clinssen) [inline doc/htmldoc/static/css/custom.css:116]: remove

File: doc/htmldoc/static/css/custom.css [hunk 4]
```diff
@@ -190,18 +196,26 @@ div[role="main"] .md-source {
...
 /* settings for icons */
-.md-typeset .sd-card-title img {
+.md-typeset .sd-card-title img, .md-typeset .sd-card-header img{
   border: none;
-  opacity: 0.75;
-  width: 64px;
-  height: auto;
-  margin-right: 0.7em;
+  opacity: 0.9;
+  /*width: 64px;*/
```
Comment (clinssen) [inline doc/htmldoc/static/css/custom.css:206]: remove

File: doc/htmldoc/static/img/arrow_left.svg
Comment (clinssen) [inline doc/htmldoc/static/img/arrow_left.svg:1]: looks like a foreign object exists in the top right

**Comment by clinssen** [comment]
Cheers!

Should this point still be handled in this PR? "TODO: Add description to diagram"

**Comment by jessica-mitchell** [comment]
@clinssen thanks for your initial comments, I removed the commented out lines, and fixed the svg

**Comment by jessica-mitchell** [comment]
Here are the following changes that were made after our mini doc-a-thon. Thanks for all your feedback
@steffengraber @heplesser @clinssen @ddahmen 

See [output here](https://nest-simulator--3066.org.readthedocs.build/en/3066/)

Changes made:

- [x] Add attribution link
- [x] Add acknowledgments to community
- [x] remove 3 buttons on top
- [x] reduce height of top section
- [x] change image nest::docs to nest::simulator
- [x] PyNEST API --> Python  API
- [x] Available Models --> Models
- [x] Remove conceptual approach: TO BE DISCUSSED
- [x]  Move 'cite NEST' and 'license' to USAGE
- [x] Add ebrains /  hbp logos to footer
- [x] Local toc "Contents" --> Page contents
- [x] Change title
- [x]  keep size of carousel images the same
- [x]  use indicators in carousel instead of arrows and change caption
- [ ] TO BE DISCUSSED: specific examples in carousel and/or general features (like types of plasticity)?
- [x]  Changed the name to 'NEST docs home' on breadcrumb trail to make it slightly different from 'NEST Simulator Documentation' text

What did not change, but was discussed:

- [ ] The top bar that displays 'NEST Simulator Documentation' and the global toc 'NEST  Simulator Documentation' remain as is (These two are tied to the same variable in conf.py so they can only be modified independently via CSS.). But the main reason they stay is because this top bar is responsive and the text changes on scroll to update to the current heading of visible page. And the global toc is always visible, and having this link there will allow users to always be able to click back to the home page no matter where they are on page.

**Comment by clinssen** [comment]
Hi, thanks for the updates! This looks mostly fine to me, I propose we merge this and then take it from there.

There seems to be a rendering issue on https://nest-simulator--3066.org.readthedocs.build/en/3066/attribution-list.html

Could we make the EBRAINS/HBP links on the bottom clickable? What about the EU flag?

**Comment by jessica-mitchell** [comment]
> Hi, thanks for the updates! This looks mostly fine to me, I propose we merge this and then take it from there.
> 
> There seems to be a rendering issue on https://nest-simulator--3066.org.readthedocs.build/en/3066/attribution-list.html
> 
> Could we make the EBRAINS/HBP links on the bottom clickable? What about the EU flag?

@clinssen 
I made the hbp/ebrains logo clickable, (it is a single image though so it links to ebrains.eu)
I don't know if/where the EU logo would be linked to. 
I updated the attribution-list to make it a bit nicer

**Review by heplesser — APPROVED** [review]
@jessica-mitchell Thanks a lot, this looks very nice now. I found one tiny details, see inline comment, but approve already now.

**Review by clinssen — APPROVED** [review]

**Pull request merged**